stream/stream_bstr.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of mpv.
+ *
+ * mpv is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "stream_bstr.h"
+#include "common/common.h"
+
+struct priv {
+    bstr data;
+};
+
+static int bs_fill_buffer(stream_t *s, void *buffer, int len)
+{
+    struct priv *p = s->priv;
+    bstr data = p->data;
+    if (s->pos < 0 || s->pos > data.len)
+        return 0;
+    len = MPMIN(len, data.len - s->pos);
+    memcpy(buffer, data.start + s->pos, len);
+    return len;
+}
+
+static int bs_seek(stream_t *s, int64_t newpos)
+{
+    return 1;
+}
+
+static int64_t bs_get_size(stream_t *s)
+{
+    struct priv *p = s->priv;
+    return p->data.len;
+}
+
+static void bs_close(struct stream *s)
+{
+    talloc_free(s->priv);
```

Comment:
this frees the memory for the struct (which would get freed anyway) but still not the bstr...

tip: you can save yourself the explicit free by using `talloc_steal` in open2

Reply:
Went with ta_set_parent() because de- and reconstructing the bstr would look weird.